core/commands/repo.go

```diff
@@ -87,60 +92,52 @@ order to reclaim hard disk space.
 					}
 				}
 				if errs {
-					res.SetError(fmt.Errorf("encountered errors during gc run"), cmds.ErrNormal)
+					res.SetError(fmt.Errorf("encountered errors during gc run"), cmdkit.ErrNormal)
 				}
 			} else {
 				err := corerepo.CollectResult(req.Context(), gcOutChan, func(k *cid.Cid) {
 					outChan <- &GcResult{Key: k}
 				})
 				if err != nil {
-					res.SetError(err, cmds.ErrNormal)
+					res.SetError(err, cmdkit.ErrNormal)
 				}
 			}
 		}()
 	},
 	Type: GcResult{},
-	Marshalers: cmds.MarshalerMap{
-		cmds.Text: func(res cmds.Response) (io.Reader, error) {
-			outChan, ok := res.Output().(<-chan interface{})
-			if !ok {
-				return nil, u.ErrCast()
+	Marshalers: oldcmds.MarshalerMap{
+		oldcmds.Text: func(res oldcmds.Response) (io.Reader, error) {
+			v, err := unwrapOutput(res.Output())
+			if err != nil {
+				return nil, err
 			}
 
 			quiet, _, err := res.Request().Option("quiet").Bool()
 			if err != nil {
 				return nil, err
 			}
 
-			marshal := func(v interface{}) (io.Reader, error) {
-				obj, ok := v.(*GcResult)
-				if !ok {
-					return nil, u.ErrCast()
-				}
+			obj, ok := v.(*GcResult)
+			if !ok {
+				return nil, e.TypeErr(obj, v)
+			}
 
-				if obj.Error != "" {
-					fmt.Fprintf(res.Stderr(), "Error: %s\n", obj.Error)
-					return nil, nil
-				}
+			if obj.Error != "" {
+				fmt.Fprintf(res.Stderr(), "Error: %s\n", obj.Error)
+				return nil, nil
+			}
 
-				if quiet {
-					return bytes.NewBufferString(obj.Key.String() + "\n"), nil
-				} else {
-					return bytes.NewBufferString(fmt.Sprintf("removed %s\n", obj.Key)), nil
-				}
```

Comment:
I know lint complains about this, but I still find the original
```
if quiet {
	return bytes.NewBufferString(obj.Key.String() + "\n"), nil
} else {
	return bytes.NewBufferString(fmt.Sprintf("removed %s\n", obj.Key)), nil
}
```
more logical and easier to read when error conditions are not involved.

Nothing to do, just wanted to register my objection to the change. `:)`